### PR TITLE
Updated test result aggregation script and related doc

### PR DESCRIPTION
**Pull request by ieuna**

### CODE REVIEW

File: mxd/performance-tests/README.md [hunk 1]
```diff
@@ -171,7 +171,13 @@ For more information about arguments visit [help.txt](help.txt).
 #### Run all files from test-configurations folder on separate clusters
 ```./experiment_controller.sh -f test-configurations -x kind-mxd -y shoot--edc-lpt--mxd```
 
-Enjoy the test results!
+### Visualizing Test Result
+1. Install python(this script is tested with python version: 3.8.10)
+2. Run the [result aggregation](mxd-performance-evaluation/results_aggregation.py) script to visualize the result. The script accepts two arguments: 
+   1. Test result folder and
```
Comment (alexandrudanciu) [inline mxd/performance-tests/README.md:177]: Please elaborate the expected folder structure:
- Folder containing a collection of experiments -> folder of individual experiment -> metadata file and dashboard folder -> statistics file.

File: mxd/performance-tests/README.md [hunk 2]
```diff
@@ -171,7 +171,13 @@ For more information about arguments visit [help.txt](help.txt).
 #### Run all files from test-configurations folder on separate clusters
 ```./experiment_controller.sh -f test-configurations -x kind-mxd -y shoot--edc-lpt--mxd```
 
-Enjoy the test results!
+### Visualizing Test Result
+1. Install python(this script is tested with python version: 3.8.10)
```
Comment (alexandrudanciu) [inline mxd/performance-tests/README.md:175]:
```suggestion
1. Install python (this script is tested with python version: 3.8.10)
```

Comment (alexandrudanciu) [inline mxd/performance-tests/README.md:175]:
```suggestion
1. Install python(this script was tested with python version: 3.8.10)
```

File: mxd/performance-tests/README.md [hunk 3]
```diff
@@ -171,7 +171,13 @@ For more information about arguments visit [help.txt](help.txt).
 #### Run all files from test-configurations folder on separate clusters
 ```./experiment_controller.sh -f test-configurations -x kind-mxd -y shoot--edc-lpt--mxd```
 
-Enjoy the test results!
+### Visualizing Test Result
```
Comment (alexandrudanciu) [inline mxd/performance-tests/README.md:174]: Instead of test result please refer to a collection of distinct test runs.

File: mxd/performance-tests/mxd-performance-evaluation/results_aggregation.py [hunk 4]
```diff
@@ -1,85 +1,202 @@
...
+from sklearn.linear_model import LinearRegression
+
+OEM_CARS_INITIAL = 'OEM_CARS_INITIAL'
+SUPPLIER_PARTS_INITIAL = 'SUPPLIER_PARTS_INITIAL'
+OEM_PLANTS = 'OEM_PLANTS'
+SUPPLIER_PLANTS = 'SUPPLIER_PLANTS'
+SUPPLIER_FLEET_MANAGERS = 'SUPPLIER_FLEET_MANAGERS'
+ADDITIONAL_CONTRACT_DEFINITIONS_OEM = 'ADDITIONAL_CONTRACT_DEFINITIONS_OEM'
+ADDITIONAL_CONTRACT_DEFINITIONS_SUPPLIER = 'ADDITIONAL_CONTRACT_DEFINITIONS_SUPPLIER'
+OEM_QUERY_CATALOGUE = 'OEM Query Catalog'
```
Comment (alexandrudanciu) [inline mxd/performance-tests/mxd-performance-evaluation/results_aggregation.py:15]: Instead of using a constant for this operation name, maybe ask for input via CLI, i.e.:
--regression "operation name"

**Review by alexandrudanciu — COMMENTED** [review]
Please consider the proposed improvements.

File: mxd/performance-tests/mxd-performance-evaluation/results_aggregation.py [hunk 5]
```diff
@@ -1,85 +1,222 @@
...
+    return sorted_processes
+
+
+def erase_file_contents(filename):
+    if os.path.exists(filename):
+        with open(filename, 'w') as file:
+            pass
+
+
+def plot_process_data(scenarios, stats_data, output_file):
```
Comment (alexandrudanciu) [inline mxd/performance-tests/mxd-performance-evaluation/results_aggregation.py:73]: Response times are provided as ms. This fact is only explicitly stated inside the dashboard.js embedded in the JMeter HTML output. Please also specify (ms) on the y-axis of the corresponding plots.

File: mxd/performance-tests/mxd-performance-evaluation/results_aggregation.py [hunk 6]
```diff
@@ -1,85 +1,222 @@
...
+                x_axis_data.append(s)
+        trace = go.Bar(
+            x=x_axis_data,
+            y=y_axis_data
+        )
+
+        layout = go.Layout(
+            title=c,
+            xaxis=dict(title=c),
+            yaxis=dict(title='Median Response Time')
```
Comment (alexandrudanciu) [inline mxd/performance-tests/mxd-performance-evaluation/results_aggregation.py:94]:
```suggestion
            yaxis=dict(title='Median Response Time (ms)')
```

**Review by alexandrudanciu — APPROVED** [review]
Please consider the suggested improvements.
